app/docs/md/conventions/preflight.md

```diff
@@ -3,8 +3,12 @@ title: Preflight
 ---
 
 
-The purpose of the preflight file is to supply default application state.
-It can also be used to customize data based on the request per route.
+The purpose of the preflight file is to populate the server-side store with default application state.
+The preflight function is passed the request object enabling you to populate the server-side store with customized data per route.
+
+You can use the preflight file as a way to incrementally build your data layer. Start with a static version of your store data to get everything working then progress to using API functions and a database as needed.
+
+Preflight can also work as a global middleware replacement for pages that require data but you may not want to be required to write an API endpoint for. A common pattern that can be solved with preflight is needing authenticated account data on a page that doesn't have an API endpoint.
```

Comment:
I find this sentence a bit hard to read and here is my suggestion which may not be any better.

Preflight can also work as a global middleware replacement for pages that require data in common with other pages that otherwise would not require an API route. A typical pattern that can be solved with preflight is needing authenticated account data on multiple pages without writing an API endpoint for each page.

Reply:
I agree. 
can't tell you how long I struggled with that sentence. 
Your version seems to make sense. I’ll take another stab at it.